StackOverflow on tab complete for disabled commands. [1.16.5] [PAPER] [2.19.0]

### Type of bug

Error in console

### `/ess version` output

v2.19.0-dev+104-5171215

### Server startup log

irrelevant

### EssentialsX config files

irrelevant, disable a command.

### Error log (if applicable)

irrelevant

### Bug description

[Server thread/ERROR]: [Essentials] Command pm failed:
org.bukkit.command.CommandException: Unhandled exception during tab completion for command '/pm ' in plugin Essentials v2.19.0-dev+104-5171215	
at org.bukkit.command.PluginCommand.tabComplete(PluginCommand.java:141) ~[patched_1.16.5.jar:git-Paper-585]
	at com.earth2me.essentials.Essentials.onTabCompleteEssentials(Essentials.java:574) ~[?:?]
	at com.earth2me.essentials.Essentials.onTabComplete(Essentials.java:540) ~[?:?]
	at org.bukkit.command.PluginCommand.tabComplete(PluginCommand.java:141) ~[patched_1.16.5.jar:git-Paper-585]
Caused by: java.lang.StackOverflowError

### Steps to reproduce

Disable a command;
attempt to use said disabled command.
If error doesn't happen immediatly, you can try adding a plugin that listens for that spesific command.

### Expected behaviour

No errors.

### Actual behaviour

Big errors.

--- a/Essentials/src/main/java/com/earth2me/essentials/Essentials.java
+++ b/Essentials/src/main/java/com/earth2me/essentials/Essentials.java
@@ -82,6 +82,7 @@ import org.bukkit.command.BlockCommandSender;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandSender;
 import org.bukkit.command.PluginCommand;
+import org.bukkit.command.PluginIdentifiableCommand;
 import org.bukkit.command.TabCompleter;
 import org.bukkit.entity.Player;
 import org.bukkit.event.Cancellable;
@@ -571,7 +572,10 @@ public class Essentials extends JavaPlugin implements net.ess3.api.IEssentials {
             // Check for disabled commands
             if (getSettings().isCommandDisabled(commandLabel)) {
                 if (getKnownCommandsProvider().getKnownCommands().containsKey(commandLabel)) {
-                    return getKnownCommandsProvider().getKnownCommands().get(commandLabel).tabComplete(cSender, commandLabel, args);
+                    final Command newCmd = getKnownCommandsProvider().getKnownCommands().get(commandLabel);
+                    if (!(newCmd instanceof PluginIdentifiableCommand) || ((PluginIdentifiableCommand) newCmd).getPlugin() != this) {
+                        return newCmd.tabComplete(cSender, commandLabel, args);
+                    }
                 }
                 return Collections.emptyList();
             }
@@ -676,7 +680,10 @@ public class Essentials extends JavaPlugin implements net.ess3.api.IEssentials {
             // Check for disabled commands
             if (getSettings().isCommandDisabled(commandLabel)) {
                 if (getKnownCommandsProvider().getKnownCommands().containsKey(commandLabel)) {
-                    return getKnownCommandsProvider().getKnownCommands().get(commandLabel).execute(cSender, commandLabel, args);
+                    final Command newCmd = getKnownCommandsProvider().getKnownCommands().get(commandLabel);
+                    if (!(newCmd instanceof PluginIdentifiableCommand) || ((PluginIdentifiableCommand) newCmd).getPlugin() != this) {
+                        return newCmd.execute(cSender, commandLabel, args);
+                    }
                 }
                 sender.sendMessage(tl("commandDisabled", commandLabel));
                 return true;

